### PR TITLE
feat(core): new share menu

### PR DESCRIPTION
**Pull request by JimmFly**
close AF-1224 AF-1216

image

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#7838** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7838?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JimmFly and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by CatsJuice — COMMENTED** [review]
This part should use new `design system variables` that based on token via `cssVarV2`

**Review by CatsJuice — COMMENTED** [review]
This part should use new `design system variables` that based on token via `cssVarV2`

File: packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-export.tsx [hunk 1]
```diff
@@ -1,79 +1,28 @@
...
-  const isMac = environment.isBrowser && environment.isMacOs;
 
   return (
     <>
-      <div className={styles.titleContainerStyle}>
-        {t['com.affine.share-menu.ShareViaExport']()}
-      </div>
       <div className={styles.descriptionStyle}>
-        {t['com.affine.share-menu.ShareViaExportDescription']()}
+        Download a static copy of your page to share with others.
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-export.tsx]: update i18n

**Review by pengx17 — APPROVED** [review]

**Review by CatsJuice — APPROVED** [review]

**Review by EYHN — APPROVED** [review]

**Review by EYHN — CHANGES_REQUESTED** [review]
stop

File: packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx [hunk 2]
```diff
@@ -71,7 +77,7 @@ export const AffineSharePage = (props: ShareMenuProps) => {
     isSharedPage === null || sharedMode === null || baseUrl === null;
   const [showDisable, setShowDisable] = useState(false);
 
-  const currentDocMode = useLiveData(doc.primaryMode$);
+  const currentDocMode = useLiveData(editor.mode$);
 
   const mode = useMemo(() => {
```
Comment (EYHN) [inline packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx]: Can this be deleted?

File: packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx [hunk 3]
```diff
@@ -82,27 +88,24 @@ export const AffineSharePage = (props: ShareMenuProps) => {
...
-  );
+  const onOpenWorkspaceSettings = useCallback(() => {
+    setSettingModalAtom({
+      open: true,
+      activeTab: 'workspace:preference',
+      workspaceMetadata: props.workspaceMetadata,
+    });
+  }, [props.workspaceMetadata, setSettingModalAtom]);
 
   const onClickCreateLink = useAsyncCallback(async () => {
```
Comment (EYHN) [inline packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx]: rename this to onClickAnyoneReadOnlyShare

File: packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx (hunk 2 above)
Comment (JimmFly) [reply packages/frontend/core/src/components/affine/share-page-modal/share-menu/share-page.tsx]: No, we need the current mode to determine whether to display `Copy Link to Selected Frame` option

**Review by EYHN — APPROVED** [review]

**Pull request merged**